### ISSUE TITLE
Allow an module import to be defined as "types only"

### ISSUE THREAD

**Issue by rgwilton** [netmod-wg/yang-next#95]
Today, when a module imports another module it isn't made explicit whether that import is to fulfill a path or identity dependency or simply for a typedef dependency.

Whilst working on YANG packages, I've began to think that it might be helpful to allow import statements to be annotated to indicate that it only imports type definitions, making the relationship between YANG modules a bit more clear.  E.g. readers of the YANG module would be able to know that they don't need to implement the imported module.  If this filter/annotation is defined then the intention is that it should be directly equivalent to an "import-only" YANG module in YANG library.

**Comment by billwuqin** [comment]
I like this proposal. I also think we could consider to add "when" substatement under import statement to check the path dependency, e.g., if the imported data node within specific path from another module doesn't exist, such module will not be imported and implemented.

**Comment by abierman** [comment]
At first this looks easy but it is not always.
The YANG library is where import-only is specified.
This is only relevant for modules that contain abstract and real identifiers.

The use-case here is for a new module that wants to just use the abstract identifiers
from an existing module that is mixed. 

### case 1: augment and leafref

import for augment and leafref always requires the imported module to be implemented

### case 2: import for XPath

- must-stmt: any reference to imported objects expected to be implemented
- when-stmt: any reference to imported objects could be a test and not expected to be implemented

### abtract-only-stmt

```
import foo {
    prefix foo;
    abstract-only true;   // default false
}
```


abstract-only = false 
-  complete module import

abstract-only = true: abstract import only
-  features
-  identities
-  typedefs
-  groupings


### Issues

It is an error if the importing module augments the imported module.
How does this work with groupings and the import of groupings via multiple modules?
This seems too complex to enforce.

e.g.
-  module A imports B and C
-   module B imports D with true 
-  module C imports D with false


More Discuss: complexity: high, bc: high, importance: medium

**Comment by kwatsen** [comment]
I ran into the issue as well, but solved another way  (not saying it's good though!)

FWIW, I put `if-feature` statements on all top-level protocol-accessible nodes.  That way, if wanting just, e.g., identities, a server can `implement` the module within enabling the features.

**Comment by janlindblad** [comment]
I agree this could be useful for readers to understand the relation. I don't particularly like the "negative" way of describing the relationship with "-only". I'd prefer something like
```
import foo {
  prefix foo;
  subset "types groupings features identities";
}
```
with default for subset being "subset all".

**Comment by kwatsen** [comment]
Andy: what if foo imported multiple times with different prefixes, each with different "subset"?!

general agreement with problem statement
solution is questionable.

change title or open new issue to track general issue, and then close this issue
